taf/tests/test_updater/update_utils.py

```diff
@@ -256,3 +257,112 @@ def _update_expect_error():
                 assert client_repository.path.exists()
             else:
                 assert not client_repository.path.exists()
+
+
+def check_repo_sync(client_dir, origin_auth_repo):
+    """
+    Asserts that the repositories in the client directory are updated
+    to match the commits in the origin authentication repository.
+
+    Args:
+        client_dir (Path): Path to the client directory.
+        origin_auth_repo (AuthenticationRepository): The origin authentication repository.
+    """
+    # Load the target repositories from the origin auth repo
+    client_repos = {
+        repo.name: GitRepository(client_dir, repo.name)
+        for repo in load_target_repositories(origin_auth_repo).values()
+    }
+    # Add the origin authentication repository itself to the client repositories
+    client_repos[origin_auth_repo.name] = GitRepository(
+        client_dir, origin_auth_repo.name
+    )
+
+    try:
+        check_if_commits_match(client_repos, origin_auth_repo.path.parent.parent)
+    except AssertionError:
+        return False
+    assert True, "The repositories are unexpectedly in sync."
+
+
+def verify_client_repos_state(client_dir: Path, auth_repo: AuthenticationRepository):
+    """
+    Verify that the client's repositories are in the correct state.
+    This means that the target repositories in the client repo should be in sync with the origin repo,
+    and the client's auth repo should be updated to the last validated commit.
+    """
+    client_auth_repo = AuthenticationRepository(path=client_dir / auth_repo.name)
+    client_target_repos = load_target_repositories(auth_repo, library_dir=client_dir)
+
+    check_last_validated_commit(client_auth_repo.path)
+
+    successful_update = True
+    for repo_name, client_repo in client_target_repos.items():
+        origin_repo = GitRepository(auth_repo.path.parent.parent, repo_name)
+        client_commit = client_repo.head_commit_sha()
+        origin_commit = origin_repo.head_commit_sha()
+
+        if client_commit != origin_commit:
+            successful_update = False
+            break
+
+    if successful_update:
+        check_if_commits_match(client_target_repos, auth_repo.path.parent.parent)
+    else:
+        for repo_name, client_repo in client_target_repos.items():
+            origin_repo = GitRepository(auth_repo.path.parent.parent, repo_name)
+            client_commit = client_repo.head_commit_sha()
+
+            # Get the target file path in the client repo
+            target_file_path = client_repo.path / "test1.txt"
+
+            # Ensure the target file exists
+            if not target_file_path.is_file():
+                raise FileNotFoundError(
```

Comment:
You usually don't raise errors in tests. If you want the test to fail if the file does not exist, use assert:
`assert target_file_path.is_file()`